src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -307,46 +309,58 @@ def _dispatchWflow(self, wflow):
                 try:
                     pline.run(wflow)
                 except Exception as ex:
-                    msg = "%s: General error from pipeline. Workflow: %s. Error: \n%s. "
+                    msg = "%s: General error from pipeline. Workflow: %s. Error: \n%s. " % (pline.name, wflow['RequestName'], str(ex))
                     msg += "\nWill retry again in the next cycle."
-                    self.logger.exception(msg, pline.name, wflow['RequestName'], str(ex))
+                    self.logger.exception(msg)
+                    if self._isStatusAdvanceExpired(wflow, wflow['RequestStatus']):
```

Comment:
Given that this block belongs to workflows sitting in rejected/aborted-completed, I would say that there are no constraints that would keep them stuck in these statuses, hence no need for this check here.

Reply:
Removed